### PR TITLE
[win32] Fix a possible leak and a micro-optimisation.

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

File: xbmc/Util.cpp [hunk 1]
```diff
@@ -172,16 +172,18 @@ std::string GetHomePath(const std::string& strTarget, std::string strPath)
   auto bufSize = GetCurrentDirectoryW(0, nullptr);
   if (bufSize > 0)
   {
-    auto buf = new wchar_t[bufSize];
-    if (0 != GetCurrentDirectoryW(bufSize, buf))
+    std::unique_ptr<wchar_t> buf(new wchar_t[bufSize]);
```
Comment (neo1973) [inline xbmc/Util.cpp]: Shouldn't this be `std::unique_ptr<wchar_t[]>` for arrays? (It doesn't make difference for PODs I believe, but it's nicer)

Comment (ace20022) [inline xbmc/Util.cpp]: thanks!

**Comment by Paxxi** [comment]
Was there a bad return path that could leak? Using unique_ptr is obviously correct so +1

**Comment by ace20022** [comment]
> Was there a bad return path that could leak? Using unique_ptr is obviously correct so +1

The next lines are:
```
 if (IsDirectoryValidRoot(currentDirectoryW))
         return currentDirectory;
     }
   }
 
   // If we ended up here we're most likely screwed
   // we will crash in a few seconds
   return strPath;
}
```

So according to the comment, if the leak would happen, the program would exit anyway ;)

**Comment by MartijnKaijser** [comment]
Jenkins build this please

**Comment by Paxxi** [comment]
Ahh right, any failure in that method will cause our initialization to fail and we crash or exit

**Comment by MartijnKaijser** [comment]
ok to merge?

**Comment by Paxxi** [comment]
Yup

**Pull request merged**